test/integration/mecanum_drive_system.cc

```diff
@@ -0,0 +1,537 @@
+
+      // Run server and check that vehicle didn't move
+      server.Run(true, 1000, false);
+
+      EXPECT_EQ(1000u, poses.size());
+
+      for (const auto &pose : poses)
+      {
+        EXPECT_EQ(poses[0], pose);
+      }
+
+      // Get odometry messages
+      double period{1.0 / 50.0};
+      double lastMsgTime{1.0};
+      std::vector<math::Pose3d> odomPoses;
+      std::function<void(const msgs::Odometry &)> odomCb =
+        [&](const msgs::Odometry &_msg)
+        {
+          ASSERT_TRUE(_msg.has_header());
+          ASSERT_TRUE(_msg.header().has_stamp());
+
+          double msgTime =
+              static_cast<double>(_msg.header().stamp().sec()) +
+              static_cast<double>(_msg.header().stamp().nsec()) * 1e-9;
+
+          EXPECT_DOUBLE_EQ(msgTime, lastMsgTime + period);
+          lastMsgTime = msgTime;
+
+          odomPoses.push_back(msgs::Convert(_msg.pose()));
+        };
+
+      // Publish command and check that vehicle moved
+      transport::Node node;
+      auto pub = node.Advertise<msgs::Twist>(_cmdVelTopic);
+      node.Subscribe(_odomTopic, odomCb);
+
+      msgs::Twist msg;
+
+      // Avoid wheel slip by limiting acceleration (1 m/s^2)
+      // and max velocity (0.5 m/s).
+      // See <max_velocity> and <max_aceleration> parameters
+      // in "/test/worlds/diff_drive.sdf".
+      // See <min_velocity>, <min_aceleration>, <min_jerk> and
+      // <max_jerk> parameters in "/test/worlds/diff_drive.sdf".
+      test::Relay velocityRamp;
+      const int movementDirection = (forward ? 1 : -1);
+      double desiredLinVelX = movementDirection * 0.15;
+      double desiredLinVelY = movementDirection * 0.15;
+      velocityRamp.OnPreUpdate(
+          [&](const UpdateInfo &/*_info*/,
+              const EntityComponentManager &)
+          {
+            msgs::Set(msg.mutable_linear(),
+                      math::Vector3d(desiredLinVelX, desiredLinVelY, 0));
+            pub.Publish(msg);
+          });
+
+      server.AddSystem(velocityRamp.systemPtr);
+
+      server.Run(true, 3000, false);
+
+      // Poses for 4s
+      ASSERT_EQ(4000u, poses.size());
+
+      int sleep = 0;
+      int maxSleep = 30;
+      for (; odomPoses.size() < 150 && sleep < maxSleep; ++sleep)
+      {
+        std::this_thread::sleep_for(std::chrono::milliseconds(100));
+      }
+      ASSERT_NE(maxSleep, sleep);
+
+      // Odometry calculates the pose of a point that is located half way
+      // between the two wheels, not the origin of the model. For example,
+      // if the vehicle is commanded to rotate in place, the vehicle will
+      // rotate about the point half way between the two wheels, thus,
+      // the odometry position will remain zero.
+      // However, since the model origin is offset, the model position will
+      // change. To find the final pose of the model, we have to do the
+      // following similarity transformation
+      
+      math::Pose3d tOdomModel(0.554283,0,-0.325,0,0,0);
```

Comment:
This transform is correct for the differential drive model, but not for the mecanum one.  The mecanum math derivation places the origin at the center of the four wheels, which for the model in the xml is at `math::Pose3d tOdomModel(-0.2, 0., 0., 0., 0., 0.);`

This is not noticed in the test probably because no rotation is being applied to the drive.

Reply:
Hi @glpuga and thanks for the info.
I am merging your test with mine but I have some problems with rotation. 
I will try to understand more , up to now the error is : 
'
mecanum_drive_system.cc:220: Failure
Expected: (desiredAngVel * relativeMotionRot.Yaw()) > (0.0), actual: -0.570339 vs 0
'
where desiredAngVel and relativeMotionRot.Yaw() are -0.2 2.85169.
I guess I have an inconsistent rotation.
I will fix other comments for now and then I will get back here